presto-parquet/src/main/java/com/facebook/presto/parquet/ParquetTypeUtils.java

```diff
@@ -169,7 +171,7 @@ public static Type getPrestoType(TupleDomain<ColumnDescriptor> effectivePredicat
             case DOUBLE:
                 return DoubleType.DOUBLE;
             case INT32:
-                return createDecimalType(descriptor).orElse(IntegerType.INTEGER);
+                return getInt32Type(descriptor);
```

Comment:
why `INT64` and `INT32` cases are handled differently now?

Reply:
This is because Parquet has [Logical Types](https://github.com/apache/parquet-format/blob/master/LogicalTypes.md) where additional metadata in `originalType` determines how an `int32` can be interpreted. In this case, it's checking if the int32 was configured to be interpreted as a date (number of days since Unix epoch which matches the semantics of the Presto `Date` type).

Before this change if I had a query like:
```
select count(*) from my_table where date_col = DATE '2018-01-01'
```
The query would fail with a type mismatch exception about comparing a `date` value to `integer` because the logical type wasn't considered. There's a similar issue related to this here  - https://github.com/prestodb/presto/pull/11118 - which is more involved as I was only targeting the `date` pushdown currently.

I backed out the timestamp changes but the case statement for `int64` in the future would be adjusted in a similar way to consider logical types e.g. an int64 can be logically typed as a `timestamp` where the value is in epoch millis, micro or nanos.

Reply:
Looking at this, https://github.com/prestodb/presto/pull/11118 actually solves this differently by adding the `HiveType` to `RichColumnDescriptor` then determining the mapped Presto type for an `int32` based on that [here](https://github.com/prestodb/presto/pull/11118/files#diff-801acf0a26c6706e156b2a6138552c35R205)

The outcome should be the same though.

Reply:
Added some unit tests for this in case this changes in the future.